File: ports/csharp/src/gui/painting/BcsPainter.cs
using BCS.Core.Tools;

namespace BCS.Gui.Painting {
    // BcsPainter abstracts native hardware-accelerated and software drawing ops
    public class BcsPainter {
        public BcsPen Pen { get; set; }
        public BcsBrush Brush { get; set; }

        public BcsPainter() {
        }

        public void DrawRect(BcsRect rect) {
            // Dispatch native draw rect
        }

        public void DrawLine(BcsPoint p1, BcsPoint p2) {
            // Dispatch native draw line
        }
    }
}

File: ports/java/src/gui/painting/BcsPainter.java
package bcs.gui.painting;

import bcs.core.tools.BcsGeometry.BcsPoint;
import bcs.core.tools.BcsGeometry.BcsRect;

// BcsPainter abstracts native hardware-accelerated and software drawing ops
public class BcsPainter {
    private BcsPen pen;
    private BcsBrush brush;

    public void setPen(BcsPen pen) {
        this.pen = pen;
    }

    public void setBrush(BcsBrush brush) {
        this.brush = brush;
    }

    public void drawRect(BcsRect rect) {
        // Dispatch native draw rect
    }

    public void drawLine(BcsPoint p1, BcsPoint p2) {
        // Dispatch native draw line
    }
}
